Qiime
=====

Quantitative Insights Into Microbial Ecology

Essentially, an open source bioinformatics pipeline that allows analysis of
next generation sequencing data with the help of several tools (BLAST, MUSCLE,
FastTREE etc.,) and provides meaningful insights, visualizations and
interpretations at the end.


## Overview of Sequencing Technologies

1. Sanger Sequencing
2. Pyro Sequencing
3. *Next Gen* Sequencing
   a. Illumina
   b. Ion Torrent


## Questions (Away)

1. What are the major advantages of QIIME over the other rRNA
analysis tools?
2. What are the preprocessing steps in QIIME?
3. What are the different methods of OTU picking?
4. Compare alpha vs beta diversity?
5. What are the different parameters required for quality filtering?
6. What are the tools used for aligning the sequence, assigning
taxonomy in de novo OTU picking?
7. Write down the workflow of QIIME for analyzing the ecology of environmental
samples through 16S rRNA amplicon?
8. What is the necessity of microbial diversity study?
9. Why QIIME is needed?

## Questions (Class)

Total N = 27
6, 3, 2, 1, Rest - Unclassified

Find the Simpson and Shannon index for the given data.

Whole Genome Sequencing
=======================

(Questions posted in the Group)
